src/web3mail/sendEmailCampaign.ts

```diff
@@ -0,0 +1,55 @@
+import { NULL_ADDRESS } from 'iexec/utils';
+import { ValidationError } from 'yup';
+import { handleIfProtocolError, WorkflowError } from '../utils/errors.js';
+import { addressOrEnsSchema, throwIfMissing } from '../utils/validators.js';
+import { SendEmailCampaignParams, SendEmailCampaignResponse } from './types.js';
+import { DataProtectorConsumer } from './internalTypes.js';
+
+export type SendEmailCampaign = typeof sendEmailCampaign;
+
+export const sendEmailCampaign = async ({
+  dataProtector = throwIfMissing(),
+  workerpoolAddressOrEns = throwIfMissing(),
+  campaignRequest,
+}: DataProtectorConsumer &
+  SendEmailCampaignParams): Promise<SendEmailCampaignResponse> => {
+  const vWorkerpoolAddressOrEns = addressOrEnsSchema()
+    .required()
+    .label('workerpoolAddressOrEns')
+    .validateSync(workerpoolAddressOrEns);
+  if (
+    campaignRequest?.workerpool !== NULL_ADDRESS &&
+    vWorkerpoolAddressOrEns.toLowerCase() !==
+      campaignRequest.workerpool.toLowerCase()
```

Comment:
At this point campaignRequest is not validated and may be anything leading to TypeError

```
> const campaignRequest = undefined
undefined
> campaignRequest?.workerpool !== '0x...' 
true
> campaignRequest.workerpool.toLowerCase()
Uncaught TypeError: Cannot read properties of undefined (reading 'workerpool')

```

```suggestion
      campaignRequest?.workerpool?.toLowerCase()
```